python/imageutils.py

```diff
@@ -471,6 +291,132 @@ def file_access(self):
             return "w+b"
         return "r+b"
 
+
+def insert_in_dict(data_dict, split_name, remaining_mem, bytes_per_voxel, y_size, z_size):
+
+    """Insert contiguous chunks of data contained in splits as key-value pairs in dictionary where the key is the byte position
+       in the reconstructed image and the value is the flattened numpy array containing contiguous data
+
+    Keyword arguments:
+    data_dict                           : dictionary containing key-value pairs of contiguous memory chunks. Key is byte position in reconstructed image.
+    split_name                          : filename of split
+    remaining_mem                       : how much available memory remains to be used (bytes)
+    bytes_per_voxel                     : number of bytes in a voxel in the reconstructed image
+    y_size                              : first dimension size in reconstructed image
+    z_size                              : second dimension size in reconstructed image
+
+    """
+
+    print "Reading split {0}".format(split_name)
+    split_pos = split_ext(split_name)[0].split('_')
+
+
+    split = nib.load(split_name)
+    split_shape = split.header.get_data_shape()
+
+    # A bit useless for now, but will become important if the datatype of splits and the datatype of the reconstructed do not match
+    # in which we'll have to modify data type of data array (Not implemented yet)
+    bytes_per_voxel_split = split.header['bitpix']/8
+
+    split_bytes = split.header.single_vox_offset + bytes_per_voxel_split*(split_shape[0]*split_shape[1]*split_shape[2])
+
+    read_time = 0
+
+    if remaining_mem is not None:
+        remaining_mem = remaining_mem - (split_bytes)
+
+    if remaining_mem >= 0 or remaining_mem is None:
+
+        read_s = time()     
+        data = split.get_data()
+
+        if split_shape[0] == y_size and split_shape[1] == z_size:
+            data_dict[split.header.single_vox_offset + bytes_per_voxel*(int(split_pos[-3]) + (int(split_pos[-2])) * y_size + (int(split_pos[-1]))*y_size*z_size)] = data.flatten('F')
+        else:
+            for i in range(0, split_shape[2]):
+                for j in range(0, split_shape[1]):
+                    key = split.header.single_vox_offset + bytes_per_voxel*(int(split_pos[-3]) + (int(split_pos[-2]) + j) * y_size + (int(split_pos[-1]) + i)*y_size*z_size)
+                    data_dict[key] = data[..., j, i].flatten('F')
+
+        read_time =  time() - read_s
+        print "Read time: ", read_time
+
+    else:
```

Comment:
Wasn't the plan to ignore the split for that specific cluster read if it could not fit in its entirety into memory? I guess I can add an `except/finally` that will allow code to proceed as normal even if block doesn't get inserted.